### PR TITLE
Allow Other Videos Library Type on Plex to be sync

### PR DESCRIPTION
**Pull request by raknam**

### CODE REVIEW

**Comment by jasongdove** [comment]
Hey - thanks for the PR. This is a feature that a bunch of people have asked for. Since ETV natively supports "Other Video" libraries, I would expect Plex's Other Video libraries to show as media kind `Other Video` in ETV (specifically not as media kind `Movies`), with similar fallback metadata that is regularly used for smart collections. See https://ersatztv.org/docs/user-guide/local-libraries#other-videos

File: ErsatzTV.Infrastructure/Plex/PlexServerApiClient.cs [hunk 1]
```diff
@@ -52,10 +52,10 @@ public async Task<bool> Ping(PlexConnection connection, PlexServerAuthToken toke
                 await service.GetLibraries(token.AuthToken).Map(r => r.MediaContainer.Directory);
             return directory
                 // .Filter(l => l.Hidden == 0)
-                .Filter(l => l.Type.ToLowerInvariant() is "movie" or "show")
-                .Filter(
-                    l => l.Type.ToLowerInvariant() is not "movie" ||
-                         (l.Agent ?? string.Empty).ToLowerInvariant() is not "com.plexapp.agents.none")
+                //.Filter(l => l.Type.ToLowerInvariant() is "movie" or "show")
```
Comment (jasongdove) [inline ErsatzTV.Infrastructure/Plex/PlexServerApiClient.cs]: This needs to stay, as ETV still only supports "movie" and "show" libraries from Plex. Plex's other videos are "movie" libraries with a metadata agent of `com.plexapp.agents.none` (see the filter below and https://github.com/ErsatzTV/ErsatzTV/pull/801).

I would update the `Option<PlexLibrary> Project(PlexLibraryResponse response)` method to return the appropriate library kind based on the metadata agent.

File: ErsatzTV.Scanner/Application/Plex/Commands/SynchronizePlexLibraryByIdHandler.cs [hunk 2]
```diff
@@ -66,6 +66,13 @@ public class SynchronizePlexLibraryByIdHandler : IRequestHandler<SynchronizePlex
                         parameters.Library,
                         parameters.DeepScan,
                         cancellationToken),
+                LibraryMediaKind.OtherVideos =>
```
Comment (jasongdove) [inline ErsatzTV.Scanner/Application/Plex/Commands/SynchronizePlexLibraryByIdHandler.cs:72]: This is a good idea, but without the changes recommended above, this path isn't hit; library media kind is still movies. Also, a different scanner may be needed to support other video fallback metadata so it's consistent with local other video libraries.

**Review by jasongdove — CHANGES_REQUESTED** [review]

File: ErsatzTV.Infrastructure/Plex/PlexServerApiClient.cs (hunk 1 above)
Comment (raknam) [reply ErsatzTV.Infrastructure/Plex/PlexServerApiClient.cs]: Yes, it is totally understable :) I'm not enough qualified with C# to easily complete this one. I will try to look deeper to update the code as you are advising.

Comment (raknam) [reply ErsatzTV.Infrastructure/Plex/PlexServerApiClient.cs]: Ok, understood for the filter. (MV Library is Other)
image

**Comment by raknam** [comment]
Everything should be good now :) I need your help @jasongdove to create the database migration file to create the PlexOtherVideo Table ^^
```
CREATE TABLE "PlexOtherVideo" (
    "Id" INTEGER NOT NULL CONSTRAINT "PK_PlexOtherVideo" PRIMARY KEY AUTOINCREMENT,
    "Key" TEXT NULL, "Etag" TEXT NULL,
    CONSTRAINT "FK_PlexOtherVideo_OtherVideo_Id" FOREIGN KEY ("Id") REFERENCES "OtherVideo" ("Id") ON DELETE CASCADE
)
```

**Comment by raknam** [comment]
Ok, found how to write the migration file :D

**Comment by raknam** [comment]
Added the missing file indicated by the CI. It was probably a mistake on my end, it was concidered as gitignored xD

**Comment by jasongdove** [comment]
This is the fallback metadata I mentioned previously (other videos have tags added for all folders from root of library, including root folder).

https://github.com/ErsatzTV/ErsatzTV/blob/f62e841af48ca112efdd559c153c7d1c9c7fed03/ErsatzTV.Core/Metadata/FallbackMetadataProvider.cs#L303-L312

Something like this could be added to the `PlexServerApiClient.ProjectToOtherVideoMetadata` so labels and folders are both added as tags.

**Review by jasongdove — APPROVED** [review]
Tested successfully here. Thanks for this great feature!

**Pull request merged**